### ISSUE TITLE
[Bug] Create workflow + invite issues: Post signup paywall despite workspace upgraded

### ISSUE THREAD

**Issue by spolu** [dust-tt/dust#9054]
When create a workspace + inviting + bump the workspace plan your still asked to subscribe through the flow

see https://groups.google.com/a/dust.tt/g/product-archive/c/O89zxFs9rN0

Important because this is the flow we use for our enterprise users

**Comment by philipperolet** [comment]
Started investigation. Timeline
- on Dec 1, spolu creates the workspace, bumps it to Friends and family, and invites him
- On dec 2, he creates a workspace at 1:07pm, and is at the same time a member of it, with NO_PLAN (workspace now marked "legacy")
- 2mns later, at 1:09, he joins the workspace created by spolu

Here's the [query](https://metabase.dust.tt/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJkYXRhYmFzZSI6NCwibmF0aXZlIjp7InF1ZXJ5IjoiXG5TRUxFQ1RcbiAgICBcIm1cIi5cImlkXCIsXG4gICAgXCJtXCIuXCJjcmVhdGVkQXRcIixcbiAgICBcIm1cIi5cInVwZGF0ZWRBdFwiLFxuICAgIFwibVwiLlwicm9sZVwiLFxuICAgIFwibVwiLlwid29ya3NwYWNlSWRcIixcbiAgICBcIndcIi5cIm5hbWVcIiBBUyBcIndvcmtzcGFjZU5hbWVcIixcbiAgICBcIndcIi5cImNyZWF0ZWRBdFwiIEFTIFwid3NDcmVhdGVkXCIsXG4gICAgXCJtXCIuXCJzdGFydEF0XCIsXG4gICAgXCJtXCIuXCJlbmRBdFwiXG5GUk9NXG4gICAgXCJtZW1iZXJzaGlwc1wiIEFTIFwibVwiXG5KT0lOXG4gICAgXCJ1c2Vyc1wiIEFTIFwidVwiIE9OIFwibVwiLlwidXNlcklkXCIgPSBcInVcIi5cImlkXCJcbkpPSU5cbiAgICBcIndvcmtzcGFjZXNcIiBBUyBcIndcIiBPTiBcIm1cIi5cIndvcmtzcGFjZUlkXCIgPSBcIndcIi5cImlkXCJcbldIRVJFXG4gICAgXCJ1XCIuXCJlbWFpbFwiID0gJ2RhbWllbi5kZXRjaGVycnlAZ21haWwuY29tJztcbiIsInRlbXBsYXRlLXRhZ3MiOnt9fX0sImRpc3BsYXkiOiJ0YWJsZSIsInBhcmFtZXRlcnMiOltdLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fX0=) with the right creation times for memberships / workspaces

I anticipate the issue just originates in him creating his own workspace in parallel of the existing one created by spolu

Will continue investigation to confirm

**Comment by philipperolet** [comment]
**Sitrep**
TLDR: 
1. confirmed there's a confusing situation with sign-up invitation screen appearing twice (on click on invitation link, then on clic on verification link)
2. Issue there seems restricted to the choice of password auth (otherwise no verification link)
3. Issue is that both screen appear the same, which is confusing. The second one should be directly the form with first/last name & i'm a pro stuff. 
4. AFAICT the flow is otherwise correct; on the second 'sign up' clic, enter details and then get in your workspace as admin on a friend/family plan.

Tested: 
- creating a workspace, inviting a fresh user (random gmail adress) as admin and bumping the workspace
- invitation link recieved, on click: signup proposition screen. On click on sign up, login screen.
- on login screen, **if password auth chosen** => get a verification link
- on clic on link, gets same signup proposition screen as above
- clicking signup then filling up => works smoothly afterwards

Remaining:
- testing with reloading when seeing the signup proposition the second time
- testing with changing name on signup as he did;
- and trying to understand in what situation he got to change his name; unclear from his email, it would mean he got 2 signup proposition screens which doesn't seem to be the case, unless he clicked multiple times on verification/invitation links

It seems clear he created a workspace for himself; a hypothesis is that at some point he had started with password, then switched to using gmail and created a workspace, then came back to the initial auth

**Comment by philipperolet** [comment]
After further investigation, damien had 
1. created himself a workspace, with an adress damien.detcherry@amblea.fr on Nov 19, via google auth
2. had one created by spolu on Dec 1 on adress damien.detcherry@gmail.com, via user+password login
3. created one for himself on Dec 2 on adresss damien.detcherry@gmail.com, via google auth

As such he had:
- 2 entries in our auth0 (gmail + amblea) with the gmail one being dual (don't know how that's handle) since both google login and user/password
- 3 workspaces, 2 no-plans and 1 friends-family (the one created by spolu)

**Comment by philipperolet** [comment]
Furthermore, it's confirmed that when going through a regular invite, a user that did not have a pre-existing auth0 entry -- which will always happen if choosing to create a password -- will get to the invitation screen twice:
1. when clicking on the email invitation link: at that time, goes direct to invitation screen, no user is created on dust nor on auth0
2. after successfully logging in via auth0: at that time, auth0 redirects to dust.tt home, which sees a user with no dust account, creates it, and redirects to invitation screen

While not exactly the issues described, it's at the root of confusion / reloadings, so part of the cause of the 3 issues, and should be fixed

**Comment by philipperolet** [comment]
> 3. created one for himself on Dec 2 on adresss damien.detcherry@gmail.com, via user+password login
To be confirmed if that creation was him trying to login our our app mistakenly creating one for himself in addition to the existing one

**Comment by philipperolet** [comment]
**Sitrep for this issue**
While using the invitation link from spolu, a new workspace was created for damien in addition to the spolu one. Follow-up is to:
- understand if in the flow he made he might have accidentally created it, or if in our code paths there is an accidental creation;
- check why after he entered details, the authentication first directed him to the workspace he created, then to the workspace he joined (spolu one)

**Comment by philipperolet** [comment]
@spolu quick check: was this intentionally moved to Done?

**Comment by spolu** [comment]
I don’t think so since I didn’t comment, I generally comment when I close :-)

**Comment by philipperolet** [comment]
That's what I thought :) But I will now close with a comment :+1: 
The reason Damien had 2 workspaces was either:
- he created one accidentally, notably due to the confusion of our "twice signup" => this is handled by #9053
  - e.g. by clicking on sign in on the "verification email screen", or by going separately to dust.tt to sign in with his username / password
- (less likely) we created one due to a double oauth method (google + user/password) => this has been handled by https://github.com/dust-tt/tasks/issues/1755
